packages/lockfile-file/src/read.ts

```diff
@@ -32,24 +34,60 @@ export async function readWantedLockfileAndAutofixConflicts (
   opts: {
     wantedVersion?: number
     ignoreIncompatible: boolean
+    useGitBranchLockfile?: boolean
+    mergeGitBranchLockfiles?: boolean
   }
 ): Promise<{
     lockfile: Lockfile | null
     hadConflicts: boolean
   }> {
-  const lockfilePath = path.join(pkgPath, WANTED_LOCKFILE)
-  return _read(lockfilePath, pkgPath, { ...opts, autofixMergeConflicts: true })
+  const lockfileNames: string[] = [WANTED_LOCKFILE]
+  if (opts.useGitBranchLockfile) {
+    const gitBranchLockfileName: string = await getWantedLockfileName(opts)
+    if (gitBranchLockfileName !== WANTED_LOCKFILE) {
+      lockfileNames.unshift(gitBranchLockfileName)
+    }
+  }
+  let result: { lockfile: Lockfile | null, hadConflicts: boolean } = { lockfile: null, hadConflicts: false }
+  for (const lockfileName of lockfileNames) {
+    result = await _read(path.join(pkgPath, lockfileName), pkgPath, { ...opts, autofixMergeConflicts: true })
+    if (result.lockfile) {
+      if (opts.mergeGitBranchLockfiles) {
+        result.lockfile = await _mergeGitBranchLockfiles(result.lockfile, pkgPath, pkgPath, opts)
+      }
+      break
+    }
+  }
+  return result
 }
 
 export async function readWantedLockfile (
   pkgPath: string,
   opts: {
     wantedVersion?: number
     ignoreIncompatible: boolean
+    useGitBranchLockfile?: boolean
+    mergeGitBranchLockfiles?: boolean
   }
 ): Promise<Lockfile | null> {
-  const lockfilePath = path.join(pkgPath, WANTED_LOCKFILE)
-  return (await _read(lockfilePath, pkgPath, opts)).lockfile
+  const lockfileNames: string[] = [WANTED_LOCKFILE]
+  if (opts.useGitBranchLockfile) {
+    const gitBranchLockfileName: string = await getWantedLockfileName(opts)
+    if (gitBranchLockfileName !== WANTED_LOCKFILE) {
+      lockfileNames.unshift(gitBranchLockfileName)
+    }
+  }
+  let lockfile: Lockfile | null = null
+  for (const lockfileName of lockfileNames) {
+    lockfile = (await _read(path.join(pkgPath, lockfileName), pkgPath, opts)).lockfile
+    if (lockfile) {
+      if (opts.mergeGitBranchLockfiles) {
+        lockfile = await _mergeGitBranchLockfiles(lockfile, pkgPath, pkgPath, opts)
+      }
+      break
+    }
+  }
+  return lockfile
```

Comment:
there is code duplication in this function. Reuse the logic from the function above.